src/App.tsx

```diff
@@ -1,61 +1,117 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { getUserById } from './services/user';
+
+const todosWithUserId: Todo[] = todosFromServer.map(todo => {
+  const user = getUserById(todo.userId);
+
+  return { ...todo, user };
+});
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUserId);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const maxId = todos.length > 0 ? Math.max(...todos.map(todo => todo.id)) : 0;
+
+  const onAdd = (newTodo: Todo) => {
+    const currentTodo = {
+      ...newTodo,
+      id: maxId + 1,
+    };
+
+    setTodos(currentTodos => [...currentTodos, currentTodo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onAdd({
+      id: 0,
+      title: title,
+      completed: false,
+      userId: userId,
+      user: getUserById(userId),
```

Comment:
Similar to the earlier comment, ensure that `getUserById(userId)` is handled correctly if it returns `null`. This is important to prevent passing `null` values to components that may not expect them.